packages/pluginutils/src/attachScopes.ts

```diff
@@ -0,0 +1,131 @@
+import { Node, walk } from 'estree-walker';
+
+import { AttachedScope, AttachScopes } from '../types';
+
+import extractAssignedNames from './extractAssignedNames';
+
+interface BlockDeclaration {
+  [index: string]: boolean;
+}
+
+const blockDeclarations: BlockDeclaration = {
+  const: true,
+  let: true
+};
+
+interface ScopeOptions {
+  parent?: AttachedScope;
+  block?: boolean;
+  params?: Array<Node>;
+}
+
+class Scope implements AttachedScope {
+  parent?: AttachedScope;
+  isBlockScope: boolean;
+  declarations: { [key: string]: boolean };
+
+  constructor(options: ScopeOptions = {}) {
+    this.parent = options.parent;
+    this.isBlockScope = !!options.block;
```

Comment:
Personally I think `Boolean` is easier to read but I'm not sure what's preferred in rollup's codebase :)
```suggestion
    this.isBlockScope = Boolean(options.block);
```

Reply:
Code style debate 😄  I've been using `!!` personally for eons, and it's handier than `Boolean` imho.